### ISSUE TITLE
Can't compile on W10+VS2017

### ISSUE THREAD

**Issue by DanielSik** [AlexeyAB/darknet#402]
Im trying to use the model "darknet_no_gpu" in a cpu with Opencv 3.4.0 and Visual Studio 2017, but i'm getting a error in this line of **cvdef.h**: 

`namespace cv { namespace debug_build_guard {} using namespace debug_build_guard; }
`
The error splits the next message repited 74times:
Syntax error: ";"
Syntax error: "}"
Syntax error: identifier 'cv'
...

All I did was change the libraries of the project to my libraries (instead of opencv3.0/opencv/... I have opencv/... directly; so I'm not sure if the error is because I'm using VS17/OpenCV 3.4.0, ...

Thanks in advance

**Comment by AlexeyAB** [comment]
* Do you compile `darknet_no_gpu.sln` using **x64 & Release** at the top of MSVS2017?
* Can you compile any other code using MSVS2017 and OpenCV 3.4.0?

**Comment by arielbernal** [comment]
yep it doesn't compile. I can definitely compile other code using MSV2017 and opencv from master. I have to enable compiled as c++ in order to get something compiled. But I didn't get too far since your code is c. Why do you have c++ dependencies?

**Comment by DanielSik** [comment]
Like arielbernal said, I'm using x64 & release and it doesn't work, but i can compile any other code without problems (talking about HoG+SVM, SSD examples, ...)

**Comment by phqmourao** [comment]
Hi,
I also could not compile the current version of "darknet_no_gpu.sln", although "darknet.sln" works fine.
With a previous version of Darknet from two weeks ago, I can compile both "darknet.sln" and "darknet_no_gpu.sln".

Error message:
Error	LNK2001	unresolved external symbol send_mjpeg	darknet_no_gpu	image.obj

My setup:
Windows 10 pro x64
VS 2017
OpenCV 3.4 (C:\opencv_3.0\opencv)

**Comment by AlexeyAB** [comment]
@DanielSik @phqmourao You can just open `darknet_no_gpu.sln` in MSVS and drag-n-drop files `http_stream.cpp` and `http_stream.h` from the directory `darknet/src/` to the solution browser at the left of MSVS.


I fixed it. So you just can update your code from this repo. (But I changed dependecies from CUDA 8.0 to CUDA 9.1 for `darknet.sln`)

**Comment by DanielSik** [comment]
Thanks AlexeyAB!, after following your instructions and change the folders of some input arguments, I'm able to work with the darknet_no_gpu project.
Now i will try to improve the speed of the results, cause it needs like 7 seconds per frame to work.

**Comment by AlexeyAB** [comment]
@DanielSik What CPU do you use?

On my CPU Core i7 6700K (4 Cores, 8 HT-logical-cores) detection takes ~1 sec per frame using Darknet Yolo voc-model.
![image](https://user-images.githubusercontent.com/4096485/36666558-1b9c3670-1afc-11e8-93ab-b82a3d70be6f.png)

----


If you want to Detect on CPU without training, then you can use Yolo v2 that built-in OpenCV - it is 5x faster than in the Darknet-framework: https://github.com/opencv/opencv/blob/master/samples/dnn/yolo_object_detection.cpp

Just create in MSVS your own solution, grag-n-drop `yolo_object_detection.cpp` to the solution browser at the left of MSVS, in the MSVS add paths to your OpenCV location and compile: 

* If you have OpenCV 2.4.13 then you should set paths after `.sln`-file is opened

    * (right click on project) -> properties -> C/C++ -> General -> Additional Include Directories: `C:\opencv_2.4.13\opencv\build\include`

    * (right click on project) -> properties -> Linker -> General -> Additional Library Directories: `C:\opencv_2.4.13\opencv\build\x64\vc14\lib`

**Comment by VanitarNordic** [comment]
@AlexeyAB 

Generally speaking, can I migrate from VS-2015 to VS-2017 and work with this repo without any problem? or do you suggest to stick with VS-2015?

**Comment by AlexeyAB** [comment]
@VanitarNordic I didn't try it. but I think yes you can. 
But you should use OpenCV 3.4.0 or lower.
OpenCV 3.4.1 and current master branch of OpenCV doesn't support C API: https://github.com/AlexeyAB/darknet/issues/414#issuecomment-369409876

**Comment by VanitarNordic** [comment]
@AlexeyAB 

So that issue with OpenCV-3.4.1 only shows itself with VS-2017? or anyway it happens with VS-2015 either?

**Comment by AlexeyAB** [comment]
@VanitarNordic Issue with `cvdef.h` happens with any MSVS 2015/2017 or any GCC on Linux, if you use OpenCV 3.4.1 or master branch.

**Comment by wuruijie123** [comment]
@DanielSik I meet this problem, because c has not 'namespace' (it belong to c++)
I worked out the problem:
1、download use opencv3.0.0 source code
2、build opencv3.0.0 with vs2015 (you may be use vs2017) ，remember don't use cuda, it spends too long  time
3、if on Debug, you must link opencv_world300d.lib,(not opencv_world300.lib)
modify detector.c, line 25:
   #ifdef _DEBUG
   #pragma comment(lib, "opencv_world" OPENCV_VERSION "d.lib")
   #else
   #pragma comment(lib, "opencv_world" OPENCV_VERSION ".lib")
   #endif
4、build opencv_world300d.lib / opencv_world300.lib may cause some problems，you can refer to 
https://stackoverflow.com/questions/38266085/how-do-i-workaround-error-c1001-in-visual-c-compiler

**Comment by petetakacs** [comment]
Hi, 

I have a compile error also on darknet_no_gpu. "The code execution cannot proceed because OpenCV_world340.dll was not found." OpenCV is installed in C:\opencv_3.0.

MSVS: 2017
Windows 10 x64

Hope you guys can help me!
